R/XcmsExperiment.R

```diff
@@ -1357,6 +1357,68 @@ setMethod(
         object
 })
 
+#'@rdname adjustRtime
+setMethod(
+    "adjustRtime", signature(object = "XcmsExperiment", param = "LamaParama"),
+    function(object, param, BPPARAM = bpparam(), ...) {
+        if(!hasChromPeaks(object))
+            stop("'object' needs to have detected chromPeaks. ",
+                 "Run 'findChromPeaks()' first")
+        if (hasAdjustedRtime(object))
+            stop("Alignment results already present. Please either remove ",
+                 "them with 'dropAdjustedRtime' in order to perform an ",
+                 "alternative, new, alignment, or use 'applyAdjustedRtime'",
+                 " prior 'adjustRtime' to perform a second round of ",
+                 "alignment.")
+        fidx <- as.factor(fromFile(object))
+        rt_raw <- split(rtime(object), fidx)
+        cp_raw  <- split.data.frame(chromPeaks(object)[, c("mz", "rt")],
+                                    chromPeaks(object)[, "sample"])
+        idx <- seq_along(object)
+
+        # Check if user as ran matching lama vs chrompeaks beforehand
+        if(length(param@rtMap) == 0)
+            param <- matchLamasChromPeaks(object, param)
+        rtMap <- param@rtMap
+        if(length(rtMap) != length(object))
+            stop("Mismatch between the number of files matched to lamas: ",
+                 length(rtMap), "and files in the object: ", length(object))
+
+        # Make model and adjust retention for each file
+        rt_adj <- bpmapply(rtMap, rt_raw, idx, FUN = function(x, y, i, param) {
+            if (nrow(x) >= 10) { # too strict ? Gam always throws error when less than that and loess does not work that well either.
```

Comment:
This is the check where I'm not completely sure (see my comment below). COuld be very dataset specific. This is what I decided based on my test datasets results.